### PR TITLE
Adds ysld unsupported module.

### PR DESCRIPTION
**Pull request by jdeolive**
YSLD is a Yaml based representation of YSLD with some syntactic nicities added to the mix. The module was developed by Boundless and is being contributed to the community. Thanks Boundless!

### CODE REVIEW

File: modules/unsupported/ysld/pom.xml [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+  
+  <parent>
+    <groupId>org.geotools</groupId>
+    <artifactId>unsupported</artifactId>
+    <version>15-SNAPSHOT</version>
+  </parent>
+  
+  <groupId>org.geotools</groupId>
+  <artifactId>gt-ysld</artifactId>
+  <name>Ysld Parser/Encoder</name>
```
Comment (jodygarnett) [inline modules/unsupported/ysld/pom.xml]: Ylsd or YSLD :)

File: modules/unsupported/ysld/pom.xml [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    </dependency>
+    <dependency>
+      <groupId>org.geotools</groupId>
+      <artifactId>gt-render</artifactId>
+      <version>${project.version}</version>
+    </dependency>
+    <dependency>
+      <groupId>org.yaml</groupId>
+      <artifactId>snakeyaml</artifactId>
+      <version>1.13</version>
```
Comment (jodygarnett) [inline modules/unsupported/ysld/pom.xml]: Use of dependency management for version numbers, especially for things like hamcrest and findbugs that may conflict.

File: modules/unsupported/ysld/src/main/java/org/geotools/ysld/Band.java [hunk 3]
```diff
@@ -0,0 +1,92 @@
+/*
+ *    GeoTools - The Open Source Java GIS Toolkit
+ *    http://geotools.org
+ *
+ *    (C) 2014-2016, Open Source Geospatial Foundation (OSGeo)
+ *
```
Comment (jodygarnett) [inline modules/unsupported/ysld/src/main/java/org/geotools/ysld/Band.java]: This is one for @smithkm - should we have a (C) 2015-2016 Boundless to reflect this code has shipped in another product?

**Comment by jdeolive** [comment]
Thanks @jodygarnett. Changed Ysld to YSLD (assuming that is the decided upon convention), added kevin and myself and module maintainers and moved dependency versions into dependency management section.

**Comment by jodygarnett** [comment]
Added an updated readme, and changed the headers to show file history.

File: modules/unsupported/ysld/src/main/java/org/geotools/ysld/Band.java (hunk 3 above)
Comment (smithkm) [inline modules/unsupported/ysld/src/main/java/org/geotools/ysld/Band.java]: @jodygarnett I guess that comes down to that argument you had with Arne Kepp about the GWC headers.  I have no idea which is the correct way to do this to make the lawyers happy.

**Pull request merged**